src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/SecurityUtils.cs

```diff
@@ -123,6 +154,11 @@ internal static void ResetAllCertificates(X509Certificate2Collection certificate
             }
         }
 
+        internal static bool TryCreateX509CertificateFromRawData(byte[] v, out X509Certificate2 certificate)
+        {
```

Comment:
Why would you throw instead of porting the code? It's trivial to port, and this functionality is needed. A client can send a base64 encoded public certificate as part of the security token and this method is used to convert that into an X509Certificate2 instance.